File: Neetcode/ValidateBST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        smallnode = []
        def lrrt(root):
            if root is None:
                return None
            lrrt(root.left)
            if smallnode:
                if smallnode[0][0] < root.val:
                    smallnode[0] = (root.val, smallnode[0][1] and True)
                else:
                    smallnode[0] = (root.val, False)
            else:
                smallnode.append((root.val, True))
            print(smallnode[0])
            lrrt(root.right)
        lrrt(root)
        return smallnode[0][1]

File: Neetcode/ValidateBST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> s;

    void lrri(TreeNode* node){
            if (node == nullptr){
                return;
            }
            lrri(node->left);
            s.push_back(node->val);
            lrri(node->right);
        }
    
    bool isValidBST(TreeNode* root) {
        lrri(root);
        for (int i=1; i<s.size(); i++){
            if (s[i-1] >= s[i]){
                return false;
            }
        }
        return true;
    }
};
